### PR TITLE
[WIP]

### PR DESCRIPTION
**Pull request by jschwinger23**

### CODE REVIEW

**Comment by jschwinger23** [comment]
remaining tasks:

1. grpc 联调
2. container feedback:
  1. event
  2. timer
  3. load && crash
3. log attach
  1. API
  2. coroutine
4. misc
  1. signal
  2. CI/CD
  3. sentry

**Comment by iwinux** [comment]
前排围观

File: src/model/node.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    pub fn get_current() -> Self {
+        let orche = get_orchestrator();
+        let conf = config::Config::get();
+        orche.get_node(&conf.hostname)
+    }
+
+    pub fn ip_address(&self) -> String {
+        let re = Regex::new(r"tcp://(?P<ip>[^:]+):(?P<port>\d+)").unwrap();
+        let caps = re.captures(&self.endpoint).unwrap();
+        caps.name("ip").unwrap().as_str().to_string()
```
Comment (iwinux) [inline src/model/node.rs]: 这个 endpoint 是外部输入吗，panic on the way...

File: src/libs/cache.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            None => None,
+        }
+    }
+
+    pub fn set(&self, id: String, healthy: Healthy) {
+        self.cache.lock().unwrap().insert(id, healthy);
+    }
+}
+
+pub fn get_cache() -> &'static HealthCache {
```
Comment (iwinux) [inline src/libs/cache.rs]: 能用 lazy_static 么

File: src/logging.rs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+pub fn init() {
+    let conf = Config::get();
+    let level = match conf.log_level.to_lowercase().as_str() {
+        "panic" | "fatal" => slog::Level::Critical,
+        "error" => slog::Level::Error,
+        "warn" | "warning" => slog::Level::Warning,
+        "info" => slog::Level::Info,
+        "debug" => slog::Level::Debug,
+        "trace" => slog::Level::Trace,
+        _ => panic!("invalid log level"),
```
Comment (iwinux) [inline src/logging.rs]: 这里可以 `slog::Level::from(&conf.log_level.to_lowercase())`?

Ref: https://docs.rs/slog/2.5.2/src/slog/lib.rs.html#2199-2206

**Pull request closed without merging**